sorting/tim_sort/tim_sort.cpp

```diff
@@ -0,0 +1,135 @@
+/*
+    Cpp program to perform Tim Sort
+
+    Algorithm
+
+    Step 1 - Divide the array into the number of blocks known as run.
+    Step 2 - Consider the size of run, either 32 or 64.
+    Step 3 - Sort the individual elements of every run one by one using insertion sort.
+    Step 4 - Merge the sorted runs one by one using the merge function of merge sort.
+    Step 5 - Double the size of merged sub-arrays after every iteration.
+
+    Time Complexity Analaysis 
+
+    Best    Case O(n)
+    Average Case O(n*log(n))
+    Worst   Case O(n*log(n))
+
+    Space O(n)
+
+*/
+#include <iostream>
+using namespace std;
+
+const int RUN_SIZE = 32; // for larger data sets we can use 64 as the run size
+
+int getMin(int x, int y)
+{
+    return (x < y) ? x : y;
+}
+
+// insertion sort
+void perfInsSort(int arr[], int start, int stop)
+{
+    for (int idx = start + 1; idx <= stop; ++idx)
+    {
+        int current = arr[idx];
+        int prev = idx - 1;
+        while (prev >= 0 && arr[prev] > current)
```

Comment:
This logic seems to be wrong.....try for all edge cases ( array sizes more than 32 or 64 ) and comment an output screenshot below this pr comment

Reply:
my bad it should be prev>=start
will change it and commit

Reply:
Yes, good eyes....keep it up